ontobio/io/qc.py

```diff
@@ -352,15 +352,16 @@ def test(self, annotation: association.GoAssociation, config: assocparser.AssocP
 class GoRule16(GoRule):
 
     def __init__(self):
-        super().__init__("GORULE:0000016", "All IC annotations should include a GO ID in the \"With/From\" column", FailMode.SOFT)
+        super().__init__("GORULE:0000016", "All IC annotations should include a GO ID in the \"With/From\" column that is also different from the entry in the \"GO ID\" column", FailMode.SOFT)
 
     def test(self, annotation: association.GoAssociation, config: assocparser.AssocParserConfig, group=None) -> TestResult:
         evidence = str(annotation.evidence.type)
         withfrom = annotation.evidence.with_support_from
+        goId = annotation.object.id
 
         okay = True
         if evidence == ic_eco:
-            only_go = [t for conjunctions in withfrom for t in conjunctions.elements if t.namespace == "GO"] # Filter terms that aren't GO terms
+            only_go = [t for conjunctions in withfrom for t in conjunctions.elements if (t.namespace == "GO" and goId is not None and t.identity != goId.identity)] # Filter terms that aren't GO terms and different from GO ID
```

Comment:
Is the `goId is not None` criteria necessary here? If `annotation.object.id` is blank this annotation is already in pretty bad shape and would definitely fail/crash somewhere else.

Reply:
ok